views/components/MovingCamera.jsx

```diff
@@ -1,33 +1,45 @@
-import React, { PropTypes } from 'react'
-var ReactTHREE = require('react-three')
-import { cameraRotation } from '../actions/Camera'
-var PerspectiveCamera = ReactTHREE.PerspectiveCamera;
-
-let dispatcher;
-document.onkeypress = (oPEvt)=> { //shouldnt this be on the container? =/
-    if(!dispatcher)
-        return;
-    var oEvent = oPEvt || window.event, charCode = oEvent.charCode;
-    if (charCode == 105 || charCode == 106 || charCode == 107 || charCode == 108 || charCode == 13 )
-        dispatcher(cameraRotation(charCode))
-};
+import React from 'react';
+import { cameraRotation } from '../actions/Camera';
+import * as BrowserActions from '../actions/Browser';
+
+const ReactTHREE = require('react-three');
+
+const PerspectiveCamera = ReactTHREE.PerspectiveCamera;
+
+class MovingCamera extends React.Component {
+  componentDidMount = () => {
+    this.props.dispatch(
+      BrowserActions.addEventListener('MovingCamera', 'keypress', this.rotateCamera)
+    );
```

Comment:
Might be a good idea, but later one when we use a middleware like `redux-thunk` you will see that it auto connects the dispatcher for you and it's a common practice for components to dispatch actions or actions creators.